lib/events.js

```diff
@@ -98,6 +98,59 @@ EventEmitter.prototype.getMaxListeners = function getMaxListeners() {
   return $getMaxListeners(this);
 };
 
+// Returns the longest sequence of `a` that fully appears in `b`,
+// of length at least 3.
+// This is a lazy approach but should work well enough, given that stack
+// frames are usually unequal or otherwise appear in groups, and that
+// we only run this code in case of an unhandled exception.
+function longestSeqContainedIn(a, b) {
+  for (var len = a.length; len >= 3; --len) {
+    for (var i = 0; i < a.length - len; ++i) {
+      // Attempt to find a[i:i+len] in b
+      for (var j = 0; j < b.length - len; ++j) {
+        let matches = true;
+        for (var k = 0; k < len; ++k) {
+          if (a[i + k] !== b[j + k]) {
+            matches = false;
+            break;
+          }
+        }
+        if (matches)
+          return [ len, i, j ];
+      }
+    }
+  }
+
+  return [ 0, 0, 0 ];
+}
+
+function longestCommonSubsequence(a, b) {
+  const [ l1, i1, j1 ] = longestSeqContainedIn(a, b);
+  const [ l2, i2, j2 ] = longestSeqContainedIn(b, a);
+  if (l1 > l2)
+    return [ l1, j1, i1 ];
+  else
+    return [ l2, i2, j2 ];
+}
+
+function enhanceStackTrace(err, own) {
+  const sep = '\nEmitted \'error\' event at at:\n';
+
+  const errStack = err.stack.split('\n').slice(1);
+  const ownStack = own.stack.split('\n').slice(1);
+
+  const [ len, off ] = longestCommonSubsequence(errStack, ownStack);
+  if (len > 0) {
+    ownStack.splice(off + 1, len - 1,
+                    '    [... lines matching original stack trace ...]');
+  }
+  // Do this last, because it is the only operation with side effects.
+  err.stack = err.stack + sep + ownStack.join('\n');
+}
+
+const kExpandStack = Symbol('kExpandStack');
+Object.defineProperty(EventEmitter, 'kExpandStack', { value: kExpandStack });
```

Comment:
is that made public on purpose?

Reply:
@targos It’s non-enumerable, so most people will never find out it exists… I don’t think anybody is going to use it? How would we hide it?

Reply:
Could it be exported by an internal module? Or is it too early in the bootstrapping process to do it?

Reply:
@targos It’s fine, I’ve changes it – the worst case scenario is that this mechanism just doesn’t work and leaves the error unchanged.